relayer/pkg/starknet/client.go

```diff
@@ -49,18 +49,18 @@ type ReaderWriter interface {
 
 var _ ReaderWriter = (*Client)(nil)
 
-var _ caigotypes.Provider = (*Client)(nil)
+// var _ caigotypes.Provider = (*Client)(nil)
 
 type Client struct {
-	gw             *caigogw.GatewayProvider
+	Gw             *caigogw.GatewayProvider
```

Comment:
just a thought, maybe rather than exporting it directly, you can have a method to export it? no big deal tho